src/plus/gk/serverConnection.ts

```diff
@@ -21,8 +16,30 @@ interface GKFetchOptions extends FetchOptions {
 	organizationId?: string | false;
 }
 
+class GKRetriever implements Retriever {
+	id = 'gitkraken';
+	name = 'GitKraken';
+	statusPageUrl = 'https://cloud.gitkrakenstatus.com/';
+
+	constructor(
+		public token: string | undefined,
+		private subscriptionService: SubscriptionService,
+	) {}
+
+	trackRequestException(): void {
+		this.subscriptionService.trackRequestException();
+	}
+
+	resetRequestExceptionCount(): void {
+		this.subscriptionService.resetRequestExceptionCount();
+	}
```

Comment:
Could these functions, and properties like `requestExceptionCount` and `requestsAreBlocked` be defined here rather than in the subscription service? If we want callers to be able to reset the exception count through this service, or to access the exception count through this service, we can define and expose those properties/functions here. Then for example, subscription service can check `this.connection.requestsAreBlocked` etc.

Reply:
done